docs/testing_mbed_OS_5.md

```diff
@@ -0,0 +1,187 @@
+# Testing in mbed OS 5
+
+The way tests are ran and compiled in mbed OS 5 is substantially different than in previous versions of mbed. Previously tests were located in one known location and a python file was kept (`tools/tests.py`) keeping track of their dependencies, capabilities, and configurations. mbed OS 5 has adopted a more distributed approach to testing. Now test code lives alongside the test code, and these are dynamically discovered by the test tools.
+
+## Table of Contents
+
+- [Using tests](#using-tests)
+  - [Test code structure](#test-code-structure)
+    - [Test discovery](#test-discovery)
+    - [Test names](#test-names)
+  - [Building tests](#building-tests)
+  - [Running tests](#running-tests)
+  - [Writing tests](#writing-tests)
+- [Debugging tests](#debugging-tests)
+- [Known issues](#known-issues)
+
+## Test code structure
+
+Tests can exist throughout mbed OS and your project's code. They are located under a special directory called `TESTS` (case is important!).
+
+Placing code under this directory means it will be ignored when building applications and libraries. This code is only ever used when building tests. This is important since all tests require a `main()` function, and building it with your application would cause multiple `main()` functions to be defined.
+
+In addition to being placed under a `TESTS` directory, test sources must exist under two other directories: a "test group" folder and a "test case" folder. The following is an example of this structure:
+```
+myproject/TESTS/test_group/test_case_1
+```
+
+In this example, `myproject` is the project root and all the source files under the `test_case_1` directory will be included in the test. Any other source files from the OS, libraries, and your project that apply to your target's configuration will also be included in the build of your test.
+
+### Test discovery
+
+Since test cases can exist throughout a project, the tools must find them in your project's file structure before building them. This is done by searching for paths that match the pattern detailed above in the [Test code structure](#test-code-structure) section.
+
+Test discovery also obeys the same rules that is used when building your project. This means that tests that are placed under a folder with a prefix like `TARGET_`, `TOOLCHAIN_`, or `FEATURE_` will only be discovered, built, and ran if your current configuration matches this prefix.
```

Comment:
You're not thinking 4th dimensionally.
It's not all about tenses as in past, present and future. Sometimes it's just a question of passive and active, for example. So tests are run, or can be run, because that's the passive form.